src/main/java/com/impinjCtrl/ReaderController.java

```diff
@@ -111,22 +121,7 @@ public void call(Object... args) {
                         System.out.println("Connected to socket: " + mApiHost + ". socket ID: " + mSocketId);
 
                         // join / register id to socket io
-                        Request req = new Request.Builder()
-                                .url(mApiHost + "/api/socket/impinj?sid=" + mSocketId)
-                                .build();
-                        mHttpClient.request(req, new Callback() {
-                            public void onFailure(Call call, IOException e) {
-                                e.printStackTrace();
-                            }
-
-                            public void onResponse(Call call, Response response) throws IOException {
-                                try {
-                                    initialReader();
-                                } catch (Exception e) {
-                                    e.printStackTrace();
-                                }
-                            }
-                        });
+                        postToApi(null);
                     }
                 }).on(EVENT_START_READER, new Emitter.Listener() {
```

Comment:
maybe we should consider to use the same listener, use args to make different actions.